s3/pkg/datastore/driver/factory.go

```diff
@@ -0,0 +1,23 @@
+package driver
+
+import (
+	backendpb "github.com/opensds/multi-cloud/backend/proto"
+	exp "github.com/opensds/multi-cloud/s3/pkg/exception"
+)
+
+type DriverFactory interface {
+	CreateDriver(detail *backendpb.BackendDetail) (StorageDriver, error)
+}
+
+var driverFactoryMgr = make(map[string]DriverFactory)
+
+func RegisterDriverFactory(driverType string, factory DriverFactory) {
```

Comment:
please consider thread safety here

Reply:
Hi rhsakarpos, RegisterDriverFactory will be called in init func of the module, so , i think it is thread safe now.